### PR TITLE
Add switches for simple domain masquerade

### PR DESCRIPTION
**Pull request by jcpunk**
#### Pull Request (PR) description
When setting up simple mail routing, it is handy to have a quick way to setup domain masquerade to convert messages for DKIM or SPF acceptance.

#### This Pull Request (PR) fixes the following issues
None

### CODE REVIEW

**Comment by jcpunk** [comment]
Looks like I'll need some help with the tests

File: manifests/satellite.pp [hunk 1]
```diff
@@ -34,11 +37,17 @@
   $_mydestination = pick($mydestination, $postfix::mydestination)
   $_mynetworks = pick($mynetworks, $postfix::mynetworks)
   $_relayhost = pick($relayhost, $postfix::relayhost)
+  $_masquerade_classes    = pick($masquerade_classes, $postfix::masquerade_classes)
+  $_masquerade_domains    = pick($masquerade_domains, $postfix::masquerade_domains)
+  $_masquerade_exceptions = pick($masquerade_exceptions, $postfix::masquerade_exceptions)
```
Comment (smortex) [inline manifests/satellite.pp:42]: `pick()` will fail if all it's arguments are `undef`.  Prefer [pick_default()](https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/REFERENCE.md#pick_default-1), this should help with CI failures:

> Contrary to the pick() function, the pick_default does not fail if all arguments are empty. This allows pick_default to use an empty value as default.

Comment (jcpunk) [reply manifests/satellite.pp]: I've switched to `pick_default`, but I fear I'll still need a bit of help with the tests....

**Review by smortex — COMMENTED** [review]
We are almost there!  More hints for CI and a few minor remarks.

File: manifests/mta.pp [hunk 2]
```diff
@@ -29,12 +32,18 @@
   Optional[Pattern[/^\S+(?:,\s*\S+)*$/]]                $mydestination = undef,
   Optional[Pattern[/^(?:\S+?(?:(?:,\s+)|(?:\s+))?)*$/]] $mynetworks    = undef,
   Optional[Pattern[/^\S+$/]]                            $relayhost     = undef,
+  Optional[String]                                      $masquerade_classes  = undef,
+  Optional[String]                                      $masquerade_domains  = undef,
+  Optional[String]                                      $masquerade_exceptions = undef,
```
[the remark this answers is not in the document]
Comment (jcpunk) [reply manifests/mta.pp]: I've tried to switch over to arrays...

**Review by smortex — APPROVED** [review]
Awesome!  If you want, you can duplicate the parameters description in the comments above the classes so that it makes the life of the future person who wants to use puppet-string to manage the module documentation easier.  In any case, i am fine with this :+1:

**Comment by jcpunk** [comment]
Any chance for a merge?  I've been using it locally with this change for a bit now.

**Comment by smortex** [comment]
Nobody seems to be worried about this change so let's merge it!  Thanks!

**Pull request merged**